### ISSUE TITLE
Wrong formatting character in "Corrupted weakref? weakref_ofs should be 0, but got %d (this should not happen)"

### ISSUE THREAD

**Issue by ppisar** [Sereal/Sereal#76]
When building Sereal-Encoder-3.004, compiler reports:

```
srl_encoder.c: In function 'srl_dump_sv':
srl_encoder.c:1446:9: warning: format '%d' expects argument of type 'int', but argument 2 has type 'UV' [-Wformat=]
         croak("Corrupted weakref? weakref_ofs should be 0, but got %d (this should not happen)", weakref_ofs);
         ^
```

If you cannot use proper formatting sequence for UV type, respect at least signedness. That is use %u instread of %d.

**Comment by demerphq** [comment]
On 5 January 2015 at 13:44, ppisar <notifications@github.com> wrote:

> When building Sereal-Encoder-3.004, compiler reports:
>
> srl_encoder.c: In function 'srl_dump_sv':
> srl_encoder.c:1446:9: warning: format '%d' expects argument of type 'int', but argument 2 has type 'UV' [-Wformat=]
>          croak("Corrupted weakref? weakref_ofs should be 0, but got %d (this should not happen)", weakref_ofs);
>          ^
>
> If you cannot use proper formatting sequence for UV type, respect at least
> signedness. That is use %u instread of %d.
>

Thanks, somehow I missed this warning, I will fix it to use the right
format.

Yves


-- 
perl -Mre=debug -e "/just|another|perl|hacker/"

**Comment by demerphq** [comment]
On 5 January 2015 at 13:44, ppisar <notifications@github.com> wrote:

> When building Sereal-Encoder-3.004, compiler reports:
>
> srl_encoder.c: In function 'srl_dump_sv':
> srl_encoder.c:1446:9: warning: format '%d' expects argument of type 'int', but argument 2 has type 'UV' [-Wformat=]
>          croak("Corrupted weakref? weakref_ofs should be 0, but got %d (this should not happen)", weakref_ofs);
>          ^
>
> If you cannot use proper formatting sequence for UV type, respect at least
> signedness. That is use %u instread of %d.
>

I do not see this warning when I build, can you tell me more about your
build environment?

cheers,
Yves

-- 
perl -Mre=debug -e "/just|another|perl|hacker/"

**Comment by ppisar** [comment]
On Mon, Jan 05, 2015 at 05:22:33AM -0800, Yves Orton wrote:
> I do not see this warning when I build, can you tell me more about your
> build environment?
> 
GCC 4.9.2, glibc 2.20.90, perl 5.20.1. The complete command is:

gcc -c  -I. -D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches  -m32 -march=i686 -mtune=atom -fasynchronous-unwind-tables   -DVERSION=\"3.004\" -DXS_VERSION=\"3.004\" -fPIC "-I/usr/lib/perl5/CORE"  -DNDEBUG -DHAVE_CSNAPPY -DHAVE_MINIZ Encoder.c

I can see the warning on i686 and armv7hl. I cannot see it on x86_64. So it
seems to be limited to 32-bit platforms.

-- Petr

**Comment by demerphq** [comment]
Fixed in 3.005
